haystack/components/generators/openai.py

```diff
@@ -170,12 +170,16 @@ def from_dict(cls, data: Dict[str, Any]) -> "OpenAIGenerator":
     def run(
         self,
         prompt: str,
+        system_prompt: Optional[str] = None,
         streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
         generation_kwargs: Optional[Dict[str, Any]] = None,
     ):
         """
         Invoke the text generation inference based on the provided messages and generation parameters.
 
+        :param system_prompt:
+            The system prompt to use for text generation. If this run time system prompt is omitted, the system
+            prompt, if defined at initialisation time, is used.
```

Comment:
Let's put this docstring second after `prompt` to match the ordering of the variables.